lib/seahorn/BmcPass.cc

```diff
@@ -49,6 +50,17 @@ static llvm::cl::opt<bool>
     ComputeCoi("horn-bmc-coi", llvm::cl::desc("Compute DataFlow-based COI"),
                llvm::cl::init(false), llvm::cl::Hidden);
 
+enum class BmcSolverKind { LEGACY, Z3, YICES2 };
```

Comment:
Don't call the current implementation `LEGACY`. It is not legacy yet, and I'm not yet convinced that we can do everything we want with the generic interface. Let's call them
`z3`, `smt-z3`, and `smt-y2` , where `smt` will stand for generic smt-lib based interface.